### DISCUSSION TITLE
fastlane match "Export Distribution Certificate" - bad magic number

### DISCUSSION THREAD

**Discussion by kaspersoerensen** [fastlane/fastlane#22223] [Q&A]
I am using fastlane *match* to generate distribution certificates and signing profiles.
I have generated my certificates with 
```fastlane match appstore```
The certificates are in the corresponding git repo - so far so good.

When I follow the steps from [the docs](https://docs.fastlane.tools/actions/match/) (section Export Distribution Certificate and Private Key as Single .p12 File), I get stuck at the first line of command:

```
openssl aes-256-cbc -k 'PASSWORD' -in "certs/distribution/L6M94C3LFA.cer" -out "cert.der" -a -d -md 
bad magic number
```
I have tried with ```-md md5``` and ```-md sha256```

I am using version 3.3 of OpenSSL. I have verified with ```fastlane env``` that ```fastlane``` also uses version 3.3. 

Do you have any good suggestions for things to try out? It is a classic scenario: it worked before (about a year ago) and now it is time to generate new certificates and I cannot figure out how to get it to work. I am sure I am missing something basic here, because the docs are nice and simple.

**Comment by kaspersoerensen** [comment]
If anyone is looking for a workaround:
use ```fastlane match distribution``` to install the certificates on your local machine. Open the keychain and export the distribution certificate from here as a .p12 file.

**Comment by fSilvaLH** [comment]
I had the same problem. The solution that worked for me was using the script provided by fastlane: match_file:

`bundle exec match_file decrypt encrypted.cer ./decrypted.cer`
